lib/elixir/lib/exception.ex

```diff
@@ -1576,6 +1576,11 @@ defmodule UndefinedFunctionError do
     {"function #{formatted_fun} is undefined or private", true}
   end
 
+  defp message(:"undefined local", nil, function, arity) do
+    "nil." <> formatted_fun = Exception.format_mfa(nil, function, arity)
```

Comment:
```suggestion
    formatted_fun = Exception.format_fa(function, arity)
```

?

Reply:
Just to confirm: the current implementation of `format_fa` [assumes a function](https://github.com/elixir-lang/elixir/blob/58bb5194d3cb1c4e702b0b1b0d065b1f551b9651/lib/elixir/lib/exception.ex#L676) (not an atom), should I update it to support a function name as well?

Reply:
Is [717238c](https://github.com/elixir-lang/elixir/pull/12998/commits/717238ca3c1fd361996757901d70490871f64ace) what you had in mind, or should I revert?

Reply:
Oh, my bad. Let’s revert. But we need to have logic for this somewhere, no? What does the compiler do in those cases? Just interpolation? Let’s mirror it if we can.

Reply:
The compiler is in a different case, because it fails on the module level here:

```elixir
iex(1)> defmodule A do
...(1)> def a, do: b()
...(1)> end
** (CompileError) iex:2: undefined function b/0 (expected A to define such a function or for it to be imported, but none are available)
```

Since here, we're just evaluating the local (sth like `:erl_eval.expr({:call, 1, {:atom, 1, :y}, []},%{},:none)`), we don't have the module context.

OK, will revert!

Reply:
Sure, my question is how do we format the function in those cases. It seems we simply do `"#{name}/#{arity}"`, so that's what I would do here. :)

Reply:
Done in [da8ea1d](https://github.com/elixir-lang/elixir/pull/12998/commits/da8ea1d1bdb75a7cb9822f16582fcde1bfafe92c)